### PR TITLE
samples: nrf_desktop: Add key remapper module

### PR DESCRIPTION
**Pull request by pdunaj**
Jira:DESK-250

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

### CODE REVIEW

File: samples/nrf_desktop/src/modules/button_remapper.c [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+#include "hid_keymap.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_BLE_BOND_LOG_LEVEL);
+
+
+static bool fn_active;
+
+static u16_t remapped_key[32];
```
Comment (MarekPieta) [inline samples/nrf_desktop/src/modules/button_remapper.c]: Maybe make it as Kconfig option or #define instead of hardcoded value?

Comment (pdunaj) [reply samples/nrf_desktop/src/modules/button_remapper.c]: fixed

**Comment by pdunaj** [comment]
I have added a clean up and tested the keyboard. Function keys are remapped and both fn and fn lock work fine.

Media keys are added to report map but are not supported fully yet as we do not have any way of correctly send the report with them. I will add that as a separate task.

@pizi-nordic , @Qbicz , @jhn-nordic I have mapped the peer control button to "Page Down" + Fn.

File: samples/nrf_desktop/src/modules/fn_keys.c [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+static void validate_enabled_fn_keys(void)
+{
+	static bool done;
+
+	if (IS_ENABLED(CONFIG_ASSERT) && !done) {
+		/* Validate the order of key IDs on the fn_keys array. */
+		for (size_t i = 1; i < ARRAY_SIZE(fn_keys); i++) {
+			if (fn_keys[i - 1] >= fn_keys[i]) {
+				__ASSERT(false, "The fn_keys array must be "
```
[the remark this answers is not in the document]
Comment (pdunaj) [reply samples/nrf_desktop/src/modules/fn_keys.c]: we cannot do static assert as this is done in runtime.

**Comment by pdunaj** [comment]
Hi @pizi-nordic , can you merge?

**Pull request merged**